### PR TITLE
content assist not working for method with unresolved arguments

### LINKED ISSUE #767: content assist not working for method with unresolved arguments
Steps to reproduce:

- create this code
```
public class Foo {
	public void foo() {
		"abcde".substring(i).| 
	}
}
```
- completion on `substring(i).` fails if `i` is not resolved.

completion

A related issue - https://github.com/redhat-developer/vscode-java/issues/2922

### PR DESCRIPTION
**Pull request by snjeza**
Fixes #767

### CODE REVIEW

**Review by gayanper — APPROVED** [review]
I think this is a good improvement among lot of completions issues when there are errors in code which sometimes forces the developer to fix all just to get the completions wanted.

Lets merge this when master is open for next release since this is not a regression.

**Comment by gayanper** [comment]
@snjeza shall we merge this ?

**Comment by snjeza** [comment]
> shall we merge this ?

@gayanper Yes, we shall.
I will resolve the conflicts.

**Comment by snjeza** [comment]
@gayanper I have resolved the conflicts.

File: org.eclipse.jdt.core/pom.xml [hunk 1]
```diff
@@ -17,7 +17,7 @@
     <version>4.28.0-SNAPSHOT</version>
   </parent>
   <artifactId>org.eclipse.jdt.core</artifactId>
-  <version>3.33.100-SNAPSHOT</version>
+  <version>3.33.101-SNAPSHOT</version>
```
Comment (iloveeclipse) [inline org.eclipse.jdt.core/pom.xml:20]: @snjeza, @gayanper : please next time don't increment bundle version **if it is already incremented** for the current release and if not, it shouldn't be +1 but +100.

The strategy is to increment minor segment by +1 if there is a new API and service segment by +100 if this is the **first** change in the release.

**Comment by snjeza** [comment]
> please next time don't increment bundle version if it is already incremented for the current release and if not, it shouldn't be +1 but +100.

Tests failed without it.

**Comment by iloveeclipse** [comment]
> Tests failed without it.

For sure because of commit not being on top of master.

**Pull request merged**